akka-actor/src/main/scala/akka/pattern/Patterns.scala

```diff
@@ -465,24 +469,6 @@ object PatternsCS {
   def ask(selection: ActorSelection, message: Any, timeoutMillis: Long): CompletionStage[AnyRef] =
     scalaAsk(selection, message)(new Timeout(timeoutMillis, TimeUnit.MILLISECONDS)).toJava.asInstanceOf[CompletionStage[AnyRef]]
 
-  /**
-   * A variation of ask which allows to implement "replyTo" pattern by including
-   * sender reference in message.
-   *
-   * {{{
-   * final CompletionStage<Object> f = Patterns.ask(
-   *   selection,
-   *   new akka.japi.Function<ActorRef, Object> {
-   *     Object apply(ActorRef askSender) {
-   *       return new Request(askSender);
-   *     }
-   *   },
-   *   timeout);
-   * }}}
-   */
-  def ask(selection: ActorSelection, messageFactory: japi.Function[ActorRef, Any], timeoutMillis: Long): CompletionStage[AnyRef] =
```

Comment:
why was this removed and not replaced?
we have one for `Future`:
```
def ask(selection: ActorSelection, messageFactory: japi.Function[ActorRef, Any], timeoutMillis: Long): Future[AnyRef]
```

Reply:
This signature version, `timeoutMillis` was added above, but Heiko forgot to remove this one.

Reply:
Oh wait, they are in different objects. I'll fix.